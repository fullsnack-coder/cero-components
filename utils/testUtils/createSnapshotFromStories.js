import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
