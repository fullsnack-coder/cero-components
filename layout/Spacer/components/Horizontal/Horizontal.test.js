import * as stories from './Horizontal.stories'
import createSnapshotFromStories from '../../../../utils/testUtils/createSnapshotFromStories'

describe('[ layout / Horizontal ]', () => {
  describe('#Horizontal', () =>
    createSnapshotFromStories('Horizontal', stories))
})
