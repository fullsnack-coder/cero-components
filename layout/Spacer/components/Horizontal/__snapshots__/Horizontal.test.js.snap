// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ layout / Horizontal ] #Horizontal when 'Horizontal' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="spacer horizontal is-playground"
    style="height: 100px; max-height: 100px; width: 0px;"
  />
</DocumentFragment>
`;

exports[`[ layout / Horizontal ] #Horizontal when 'Horizontal' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="spacer horizontal is-playground"
    style="height: 100px; max-height: 100px; width: 0px;"
  />
  <div
    class="spacer horizontal is-playground"
    style="height: 100px; max-height: 100px; width: 5px;"
  />
  <div
    class="spacer horizontal is-playground"
    style="height: 100px; max-height: 100px; width: 15px;"
  />
  <div
    class="spacer horizontal is-playground"
    style="height: 100px; max-height: 100px; width: 20px;"
  />
  <div
    class="spacer horizontal is-playground"
    style="height: 100px; max-height: 100px; width: 40px;"
  />
</DocumentFragment>
`;
